### PR TITLE
MESH-899 Update MIPs events and state variables

### PR DESCRIPTION
**Pull request by adamdossa**

### CODE REVIEW

File: pallets/mips/src/lib.rs [hunk 1]
```diff
@@ -640,37 +679,70 @@ decl_module! {
...
+                ReferendumType::FastTracked,
             );
 
-            Self::deposit_event(RawEvent::ProposalFastTracked(index));
-            Self::close_proposal(index);
+            Self::update_proposal_state(index, ProposalState::Referendum);
+
+            // Update proposal metadata so we don't re-execute it later
+            <ProposalMetadata<T>>::mutate( |metas| {
+                let meta = metas.iter_mut().find( |meta| meta.index == index);
```
Comment (maxsam4) [inline pallets/mips/src/lib.rs:694]: Unbounded storage access. Ok for testnet v1, should revisit for v2. Perhaps add a TODO?

File: pallets/mips/src/lib.rs [hunk 2]
```diff
@@ -640,37 +679,70 @@ decl_module! {
...
+
+            let proposal_meta = MipsMetadata {
+                proposer: proposer.clone(),
+                index,
+                end: Zero::zero(),
+                url,
+                description,
+                cool_off_until: Zero::zero(),
+            };
+            <ProposalMetadata<T>>::mutate(|metadata| metadata.push(proposal_meta));
```
Comment (maxsam4) [inline pallets/mips/src/lib.rs:738]: Unbounded storage access. Ok for testnet v1, should revisit for v2. Perhaps add a TODO?

File: pallets/mips/src/lib.rs (hunk 1 above)
Comment (adamdossa) [reply pallets/mips/src/lib.rs:695]: Added TODO comment

File: pallets/mips/src/lib.rs (hunk 2 above)
Comment (adamdossa) [reply pallets/mips/src/lib.rs:739]: Added TODO comment

**Comment by adamdossa** [comment]
@fmiguelgarcia FYI - this PR is going to clash a lot w/ the treasury PR. I left a separate comment with a couple of questions on that PR - we can discuss the best approach to merging the two.

File: pallets/mips/src/lib.rs (hunk 1 above)
Comment (fmiguelgarcia) [reply pallets/mips/src/lib.rs:695]: BTW, It has been fixed in [PR 303](https://github.com/PolymathNetwork/Polymesh/pull/303/)

File: pallets/mips/src/lib.rs (hunk 2 above)
Comment (fmiguelgarcia) [reply pallets/mips/src/lib.rs:739]: BTW, It has been fixed in [PR 303](https://github.com/PolymathNetwork/Polymesh/pull/303/)

**Pull request merged**